langlib/lang.regexp/src/main/ballerina/regexp.bal

```diff
@@ -116,6 +146,16 @@ public isolated function findAll(RegExp re, string str, int startIndex = 0) retu
 # After one match is found, it looks for the next match starting where the previous
 # match ended, so the list of matches will be non-overlapping.
 #
+# ```ballerina
+# string:RegExp r = re `([bB].tt[a-z]*)`;
```

Comment:
Let's use a regex with capturing groups for `findAllGroups`

Reply:
[4ab39fc](https://github.com/ballerina-platform/ballerina-lang/pull/39096/commits/4ab39fca6922f71d8e27c8e43da56f502c86dd96)